ksql-common/src/main/java/io/confluent/ksql/util/ReservedInternalTopics.java

```diff
@@ -0,0 +1,153 @@
+  // not configurable.
+  public static final String KSQL_INTERNAL_TOPIC_PREFIX = "_confluent-ksql-";
+  public static final String KSQL_COMMAND_TOPIC_SUFFIX = "command_topic";
+  public static final String KSQL_CONFIGS_TOPIC_SUFFIX = "configs";
+
+  /**
+   * Returns the internal KSQL command topic.
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @return The command topic name.
+   */
+  public static String commandTopic(final KsqlConfig ksqlConfig) {
+    return toKsqlInternalTopic(ksqlConfig, KSQL_COMMAND_TOPIC_SUFFIX);
+  }
+
+  /**
+   * Returns the internal KSQL configs topic (used for KSQL standalone)
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @return The configurations topic name.
+   */
+  public static String configsTopic(final KsqlConfig ksqlConfig) {
+    return toKsqlInternalTopic(ksqlConfig, KSQL_CONFIGS_TOPIC_SUFFIX);
+  }
+
+  /**
+   * Returns the KSQL processing log topic.
+   * <p/>
+   * This is not an internal topic in the sense that users are intentionally meant to read from
+   * this topic to identify deserialization and other processing errors, define a KSQL stream on
+   * it, and potentially issue queries to filter from it, etc. This is why it is not prefixed in
+   * the way KSQL internal topics are.
+   *
+   * @param config The Processing log config, which is used to extract the processing topic suffix
+   * @param ksqlConfig The KSQL config, which is used to extract the KSQL service id.
+   * @return The processing log topic name.
+   */
+  public static String processingLogTopic(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig
+  ) {
+    final String topicNameConfig = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    if (topicNameConfig.equals(ProcessingLogConfig.TOPIC_NAME_NOT_SET)) {
+      return String.format(
+          "%s%s",
+          ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+          ProcessingLogConfig.TOPIC_NAME_DEFAULT_SUFFIX
+      );
+    } else {
+      return topicNameConfig;
+    }
+  }
+
+  /**
+   * Compute a name for a KSQL internal topic.
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @param topicSuffix A suffix that is appended to the topic name.
+   * @return The computed topic name.
+   */
+  private static String toKsqlInternalTopic(final KsqlConfig ksqlConfig, final String topicSuffix) {
+    return String.format(
+        "%s%s_%s",
+        KSQL_INTERNAL_TOPIC_PREFIX,
+        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+        topicSuffix
+    );
+  }
+
+  private final Pattern hiddenTopicsPattern;
+  private final Pattern readOnlyTopicsPattern;
+
+  public ReservedInternalTopics(final KsqlConfig ksqlConfig) {
+    final ProcessingLogConfig processingLogConfig =
+        new ProcessingLogConfig(ksqlConfig.getAllConfigPropsWithSecretsObfuscated());
```

Comment:
Still not convinced this is the _right_ way to create the processing log config. This is not how it's created at the moment.

e.g. can the processing log config ever have a password in it? In which case this 'withSecretsObfuscated` would fail. 

e.g. can the `ksqlConfig` passed in here include overrides users have supplied in the CLI?

At the moment, the `ProcessingLogConfig` is created in one place and then injected / passed to where it is needed. IMHO this is a better pattern than creating it in multiple places, when ever needed: the single instance pattern ensures we're using the _same_ config everywhere.

Hence, I urge you to change the code so that we're passing down the `ProcessorLogConfig` to where its needed, ideally injecting it into constructors, rather than passing it as an argument of other methods, with it finally being passed as a separate constructor argument here.